### PR TITLE
Check for java reserved words in component and project names

### PR DESCRIPTION
**Pull request by SusanRatiLane**
Issue #1336 - Projects with reserved words do no compile correctly.
YAIL reserved words are also a bad idea, but they were only
checked for component names.

Now component and project names check for both YAIL and java words

### CODE REVIEW

**Comment by ewpatton** [comment]
@SusanRatiLane It might make sense to also restrict Screen names using the same logic. See validate(String) in AddFormCommand.java.

**Comment by SusanRatiLane** [comment]
Added validation for screen names.

File: appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java [hunk 1]
```diff
@@ -985,12 +985,16 @@
   @Description("Error shown when a new component type would be the same as a component instance name")
   String sameAsComponentInstanceNameError();
 
-  @DefaultMessage("Component name cannot be any of the following: CsvUtil, Double, Float, " +
+  @DefaultMessage("Name cannot be any of the following: CsvUtil, Double, Float, " +
       "Integer, JavaCollection, JavaIterator, KawaEnvironment, Long, Short, SimpleForm, String, " +
-      "Pattern, YailList, YailNumberToString, YailRuntimeError")
+      "Pattern, YailList, YailNumberToString, YailRuntimeError, abstract, continue, for, new, " +
+          "switch, assert, default, goto, package, synchronized, boolean, do, if, private, this, break, " +
```
Comment (ewpatton) [inline appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: The spacing switches from 6 to 10 here. It should be 6 to be consistent with the rest of the sources.

Comment (SusanRatiLane) [reply appinventor/appengine/src/com/google/appinventor/client/OdeMessages.java]: Whitespace fixed, and tests added.

File: appinventor/appengine/tests/com/google/appinventor/client/youngandroid/TextValidatorsTest.java [hunk 2]
```diff
@@ -45,4 +48,13 @@ public void testComponentIdentifierFilter(){
 	      assertFalse(TextValidators.isValidComponentIdentifier(illegalComponentIdentifier));
 	    }
   }
+
+  public void testReservedNameFilter(){
+      for (String reservedWord : reservedWordNames) {
```
Comment (ewpatton) [inline appinventor/appengine/tests/com/google/appinventor/client/youngandroid/TextValidatorsTest.java]: Indentation for new blocks should be 2 spaces per the [Google Java code style](https://google.github.io/styleguide/javaguide.html).

Comment (SusanRatiLane) [reply appinventor/appengine/tests/com/google/appinventor/client/youngandroid/TextValidatorsTest.java]: Done. And Intellij defaults fixed to boot.

**Pull request merged**